### PR TITLE
PHP 7.2 and Symfony ^4.0

### LINKED ISSUE #48: Remove Dev Dependency to symfony/expression-language
Was added in #43 to use phpstan. Can be removed once phpstan/phpstan#576 has been resolved.

### LINKED ISSUE #50: Symfony 4 Dependencies
Can the Symfony dependencies please be upgraded to include the SF4 releases? Did anybody test if this bundle is compatible yet?

If I have enough time I will try to create a fork to try this bundle out in the next few days, but I can't promise anything.

### PR DESCRIPTION
**Pull request by UFOMelkor**
On travis:
 - PHP 7.2 runs against highest dependencies
 - PHP 7.1 runs against lowest dependencies

Some things need to be adjusted to get it working for symfony 4.
 - DataCollectorInterface has a reset-Method.
 - Services are private by default. This has to be adjusted mostly in tests.
 - ResolveDefinitionTemplatesPass is not available in v4.0, but ResolveChildDefinitionsPass is not available in 3.3 (tests only).

Closes #48 
Closes #50

### CODE REVIEW

**Comment by prolic** [comment]
@UFOMelkor I suggest we test PHP 7.2 with highest AND lowest dependencies and PHP 7.1 also with highest AND lowest dependencies. This will make 4 travis builds and will take a little longer, but we are more safe like this.

**Comment by UFOMelkor** [comment]
Changed it to:
 - PHP 7.2 and 7.1 each with highest and lowest dependencies
 - coverage and phpcs only on PHP 7.2 with highest dependencies

File: test/DependencyInjection/Fixture/config/yml/event_bus_routes_with_@.yml [hunk 1]
```diff
@@ -12,4 +12,5 @@ prooph_service_bus:
 
 services:
   Acme\UserListener:
-    class: ProophTest\Bundle\ServiceBus\DependencyInjection\Fixture\Model\AcmeUserEventListener
\ No newline at end of file
+    class: ProophTest\Bundle\ServiceBus\DependencyInjection\Fixture\Model\AcmeUserEventListener
+    public: true
```
Comment (mumia) [inline test/DependencyInjection/Fixture/config/yml/event_bus_routes_with_@.yml]: Missing newline

**Comment by mumia** [comment]
Any thoughts when this will be merged?

File: test/DependencyInjection/Fixture/config/yml/event_bus_routes_with_@.yml (hunk 1 above)
Comment (UFOMelkor) [reply test/DependencyInjection/Fixture/config/yml/event_bus_routes_with_@.yml]: Thank you, I added one 👍

**Comment by UFOMelkor** [comment]
ping @prooph/symfony-team

**Review by codeliner — APPROVED** [review]

**Comment by codeliner** [comment]
great job

**Pull request merged**